src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -1101,10 +1101,16 @@ protected override void Go()
                     case RegexCode.UpdateBumpalong:
                         // UpdateBumpalong should only exist in the code stream at such a point where the root
                         // of the backtracking stack contains the runtextpos from the start of this Go call. Replace
-                        // that tracking value with the current runtextpos value.
-                        runtrack![runtrack.Length - 1] = runtextpos;
-                        advance = 0;
-                        continue;
+                        // that tracking value with the current runtextpos value if it's greater.
+                        {
+                            ref int trackingpos = ref runtrack![runtrack.Length - 1];
+                            if (trackingpos < runtextpos)
```

Comment:
Shouldn't we account for right to left scenario for the interpreted engine here?

Reply:
I can add an assert that we never get here for RightToLeft.  We only ever insert UpdateBumpalong nodes when RightToLeft is not set.  For the most part, for all the tree manipulations I added in .NET 5, I didn't invest any time/energy in making sure they were correct for RightToLeft and so explicitly check for RightToLeft and skip the optimizations before doing them.

Reply:
Oh I see, I thought this particular line could be hit on a RTL scenario, but if that's not the case then I guess that's ok. Perhaps adding an Assert might not be a bad idea (since you have to solve some merge conflicts anyway) just in case we ever add UpdateBumpalong support for RTL